google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -310,27 +611,76 @@ class Page:
-        self.form_fields = [
-            FormField(documentai_formfield=form_field, document_text=document_text)
-            for form_field in self.documentai_page.form_fields
-        ]
-        self.lines = [
-            Line(documentai_line=line, document_text=document_text)
-            for line in self.documentai_page.lines
-        ]
-        self.paragraphs = [
-            Paragraph(documentai_paragraph=paragraph, document_text=document_text)
-            for paragraph in self.documentai_page.paragraphs
-        ]
-        self.blocks = [
-            Block(documentai_block=block, document_text=document_text)
-            for block in self.documentai_page.blocks
-        ]
         self.tables = [
-            Table(documentai_table=table, document_text=document_text)
-            for table in self.documentai_page.tables
+            Table(documentai_object=table, document_text=self.document_text)
+            for table in self.documentai_object.tables
         ]
+
+        self.form_fields = [
+            FormField(documentai_object=form_field, document_text=self.document_text)
+            for form_field in self.documentai_object.form_fields
+        ]
+
+        self.tokens = _get_tokens(tokens=self.documentai_object.tokens, page=self)
+        self.lines = _get_lines(lines=self.documentai_object.lines, page=self)
+        self.paragraphs = _get_paragraphs(
+            paragraphs=self.documentai_object.paragraphs, page=self
+        )
+        self.blocks = _get_blocks(
+            blocks=self.documentai_object.blocks,
+            page=self,
+        )
+
+    def to_hocr(self, title):
+        r"""Exports a string hOCR version of the documentai.Document.Page.
+
+        Args:
+            title (str):
+                Required. The title for ocr_page.
+
+        Returns:
+            str:
+                A string hOCR version of the documentai.Document.Page.
+        """
+        f = ""
+        pidx = self.documentai_object.page_number
+        f += f"<div class='ocr_page' lang='unknown' title='image \"{title}\";{self.hocr_bounding_box}'>"
+        for bidx, block in enumerate(self.blocks):
+            f += f"<span class='ocr_carea' id='block_{pidx}_{bidx}' title='{block.hocr_bounding_box}'>"
+            for paridx, paragraph in enumerate(block.paragraphs):
+                f += f"<span class='ocr_par' id='par_{pidx}_{bidx}_{paridx}' title='{paragraph.hocr_bounding_box}'>"
+                for lidx, line in enumerate(paragraph.lines):
+                    line_text = html.escape(line.text)
+                    f += f"<span class='ocr_line' id='line_{pidx}_{bidx}_{paridx}_{lidx}' title='{line.hocr_bounding_box}'>{line_text}</span>"
+                    for tidx, token in enumerate(line.tokens):
+                        word_text = html.escape(token.text)
+                        f += f"<span class='ocrx_word' id='word_{pidx}_{bidx}_{paridx}_{lidx}_{tidx}' title='{token.hocr_bounding_box}'>{word_text}</span>"
```

Comment:
To be sure, is this structure of the `id` the hOCR convention?  Let's document this in one of the docstrings if not already done.

From the perspective of the XML, having these indices recorded in the `id` does not seem to provide very much value.  Must the `id` be meaning in anyway, or can we just use a short UUID?

Reply:
I think the `id` has some meaning, both the hocr documentation and the client that will use this have the id in that format, i'll add some description about the id in the docstring.